instrumentation/src/main/java/io/opentelemetry/android/config/PersistenceConfiguration.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.config;
+
+/** Configuration for disk buffering. */
+public final class PersistenceConfiguration {
```

Comment:
I don't really have a strong opinion on having all the params straight into the main config or grouping them within "subconfigs" if you will, though I'm a bit concerned about the intuitiveness/discoverability factor for someone who's not familiarized with the lib and wants to know all there is to configure about a specific feature. Now, this doesn't necessarily have to be the only way to address that issue, we could use a common prefix for all the config names related to a single feature, but that brings other concerns about having to enforce naming rules and also probably code placement rules (having all the related config methods together in one part of the file and so on) for expanding and maintaining the config code.

Now, the reason why it's not a strong opinion, at least not for now anyway, is because this is really only useful at a larger scale, in case there are multiple cases like this one, where several params are needed for a feature. And at the moment I can't think of another feature that would need this in the future (but this is a new project so I'm guessing that several features should come soon) and also we're talking about only 2 params for the disk buffering case (although we can technically expose more config options available in the [disk buffering lib](https://github.com/open-telemetry/opentelemetry-java-contrib/tree/main/disk-buffering#configuration), it's just that I tried to keep it simple for now).

Generally speaking, I try to make things as scalable as possible from the beginning, though I must admit that sometimes it backfires and I end up adding unnecessary things so, I like having these kinds of conversations because it helps making the line between _useful-for-scalability_ and _too-early/unnecessary-optimization_ a bit clearer.